storage/simulation/main.go

```diff
@@ -18,47 +18,106 @@
 	stopper := stop.NewStopper()
 	defer stopper.Stop()
 
+	flag.Parse()
+	// Give the flags some boundaries.
+	if *startingNodes < 0 {
+		*startingNodes = 0
+	}
+	if *maxEpoch < 0 {
+		*maxEpoch = 0
+	}
+
+	// Clean the output file strings so they can be compared easily or set them
+	// to nil if there is no file.
+	if len(*actionOutputFile) > 0 {
+		*actionOutputFile = filepath.Clean(*actionOutputFile)
+	} else {
+		actionOutputFile = nil
+	}
+	if len(*epochOutputFile) > 0 {
+		*epochOutputFile = filepath.Clean(*epochOutputFile)
+	} else {
+		epochOutputFile = nil
+	}
+	*scriptInputFile = filepath.Clean(*scriptInputFile)
+
 	fmt.Printf("A simulation of the cluster's rebalancing.\n\n")
+	fmt.Printf("Maximum Epoch for simulation set to %d.\n", *maxEpoch)
+	fmt.Printf("Cluster is starting with %d nodes.\n", *startingNodes)
+	fmt.Printf("Script file is %s\n", *scriptInputFile)
 
-	// TODO(bram): Setup flags to allow filenames for these outputs. Print both
-	// to action and epoch to os.Stdout as well.
-	epochWriter := os.Stdout
-	actionWriter := os.Stdout
-	c := createCluster(stopper, 5, epochWriter, actionWriter)
+	epochWriter := io.MultiWriter(os.Stdout)
+	actionWriter := io.MultiWriter(os.Stdout)
 
-	// Split a random range 100 times.
-	for i := 0; i < 10; i++ {
-		c.splitRangeRandom()
+	// Do we have an action output file?
+	if actionOutputFile != nil && len(*actionOutputFile) > 0 {
+		actionOutputF, err := os.OpenFile(*actionOutputFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0666)
+		if err != nil {
+			log.Fatalf("Could not create or open action file:%s - %s", *actionOutputFile, err)
+			return
+		}
+		defer actionOutputF.Close()
+		fmt.Printf("Action Output will be written to %s.\n", *actionOutputFile)
+		actionWriter = io.MultiWriter(os.Stdout, actionOutputF)
+	} else {
+		fmt.Printf("Action Output will only be written to console.\n")
 	}
-	c.flush()
 
-	// TODO(bram): only flush when on manual stepping (once that enabled).
-	// Run until stable or at the 100th epoch.
-	for c.runEpoch() != true && c.epoch < 100 {
-		c.flush()
+	// Do we have an epoch output file?
+	if epochOutputFile != nil && len(*epochOutputFile) > 0 {
+		// Is it the same as the action output file?
+		if actionOutputFile != nil && *actionOutputFile == *epochOutputFile {
+			epochWriter = actionWriter
+			fmt.Printf("Epoch Output will also be written to %s.\n", *epochOutputFile)
+		} else {
+			epochOutputF, err := os.OpenFile(*epochOutputFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0666)
+			if err != nil {
+				log.Fatalf("Could not create or open epoch file:%s - %s", *epochOutputFile, err)
+				return
+			}
+			defer epochOutputF.Close()
+			fmt.Printf("Epoch Output will be written to %s.\n", *epochOutputFile)
+			epochWriter = io.MultiWriter(os.Stdout, epochOutputF)
+		}
+	} else {
+		fmt.Printf("Epoch Output will only be written to console.\n")
 	}
-	c.flush()
-	fmt.Println(c)
 
-	for i := 0; i < 70; i++ {
-		c.splitRangeLast()
+	fmt.Printf("\nParsing Script:\n")
```

Comment:
Because I want nice clean output to the console. No need for all the details log provides.